src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -843,18 +855,22 @@ static int TopicPartition_init (PyObject *self, PyObject *args,
 	const char *topic;
 	int partition = RD_KAFKA_PARTITION_UA;
 	long long offset = RD_KAFKA_OFFSET_INVALID;
+	const char *metadata = NULL;
+
 	static char *kws[] = { "topic",
 			       "partition",
 			       "offset",
+			       "metadata",
 			       NULL };
-
-	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "s|iL", kws,
```

Comment:
We typically try to keep the happy-path (non-error path) unindented, so restore the original:
```c
if (!Py..))
  return -1;

TopicPartition_setup(),..

```